src/main/java/org/example/Twitter.java

```diff
@@ -0,0 +1,461 @@
+        } else {
+            System.out.println("Incorrect login credentials.");
+        }
+
+        return currentUserId;
+    }
+
+    public static void parseJsonInfo(String filename) throws IOException, ParseException {
+        /*
+        Function takes in a json file and fills out the local dictionaries inside Twitter
+        */
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonNode = objectMapper.readTree(new File(filename));
+
+        // Fill in the tweetDict
+        tweetDict = objectMapper.readValue(jsonNode.get("tweets").traverse(), new TypeReference<LinkedHashMap<String, Tweet>>(){});
+        // Make sure to update the tweet counter to the amount we have in the script
+        tweetCounter = tweetDict.size();
+
+        // Fill in the userDict
+        userDict = objectMapper.readValue(jsonNode.get("users").traverse(), new TypeReference<LinkedHashMap<String, User>>(){});
+
+    }
+    public static void start() throws IOException, ParseException {
+        /*
+        Function starts Twitter's command line.
+         */
+
+        Scanner in = new Scanner(System.in);
+        String s = " ";
+
+        String searchVal = "automated text";
+
+//        showAllTweets();
+//        showAllUsers();
+//        showAllFollowers();
+//        showUserTimeline();
+//        makeTweet("automated text", new ArrayList<String>(Arrays.asList("#test", "help")));
+//        searchTweet(searchVal);
+
+        while(s != "quit"){
+            s = in.nextLine();
+            if(s.toLowerCase().contains("entire timeline")){
+                showAllTweets();
+
+             } else if((s.toLowerCase().contains("show all users"))){
+                 showAllUsers();
+
+             } else if((s.toLowerCase().contains("show all my followers"))){
+                 showAllFollowers();
+
+             } else if(s.toLowerCase().contains("home timeline")) {
+                 showUserTimeline();
+
+             } else if(s.toLowerCase().contains("searching")) {
+                 System.out.print("Search Input: ");
+                 String t = in.nextLine();
+                 searchTweet(t);
+
+             } else if(s.toLowerCase().contains("tweeting")) {
+                System.out.print("Text Body: ");
+                String textBody = in.nextLine();
+
+                ArrayList<String> hashtags = new ArrayList<String>();
+                while(true){
+                    System.out.println("Add hashtag: ");
+                    String answer = in.nextLine();
+                    hashtags.add(in.nextLine());
+
+                    if (!answer.contains("#")){
+                        System.out.println("End of Hash adding");
+                        break;
+                    }
+                }
+                makeTweet(textBody, hashtags);
+
+            } else {
+                s = "quit";
+            }
+        }
+        in.close();
+    }
+
+    public static ArrayList<ArrayList<String>> showAllTweets(){
```

Comment:
do you really need to return anything here? consider what happens with a large dataset wrt memory consumption

Reply:
The idea was to have output for jUnit testing. So it depends if we care for a more stable codebase or less memory consumption.